### PR TITLE
feat(paddle): Add backward method to Paddle Frontend

### LINKED ISSUE #23407: backward

### PR DESCRIPTION
**Pull request by Boghdady9**
# PR Description

Add backward method to Paddle Frontend
## Related Issue

Close #23407 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Boghdady9** [comment]
@AnnaTz 
I'm sorry for closing the last PR, but I encountered a conflict issue, so I closed it...

here, I made little changes and I tested the function on my machine before commit it, it passed all tests after many trails. I hope no more changes😁😅

**Comment by AnnaTz** [comment]
ivy-gardener

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 1]
```diff
@@ -283,6 +283,42 @@ def test_paddle__reshape(
...
+
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="backward",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+    ),
+)
+def test_paddle_backward(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:297]: Please rename the test to `test_paddle_tensor_backward` to be in line with our naming convention.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py [hunk 2]
```diff
@@ -283,6 +283,42 @@ def test_paddle__reshape(
...
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+    backend_fw,
+):
+    input_dtype, x = dtype_and_x
+    x = Tensor(x)
+    helpers.test_frontend_method(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:308]: `x = Tensor(x)` should not be needed here. I believe this is done inside `test_frontend_method` later on.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Just a couple of minor changes before we merge :)

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py (hunk 2 above)
Comment (Boghdady9) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py:308]: @AnnaTz
 I passed the five tests only after I add  x = Tensor(x), so I think it’s necessary

**Pull request closed without merging**